script/ci_build.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+mkdir -p build && cd build
+cmake .. -DFPLUS_BUILD_UNITTEST=ON
+cmake --build . --target unittest --config Release -- -j4
```

Comment:
If you want to build the tests, then this should be:
```bash
mkdir build
# Ninja is available in GitHub Actions and automatically uses all cores
cmake -S test -B build -G Ninja
cmake --build build --config Release
```

Reply:
Ah, I see what happened here, your feature branch is missing the changes from my PR. You should rebase your commits on the current master branch.

Reply:
Yes, it's already on the todo list. I'm focusing on getting the environments up first. It great if you could review a second time, once I've merged `master`.

Regarding Ninja: I can add for the CI build, should we use it instead?

Reply:
It's not necessary, just more things to do in the CI which isn't really necessary. If the build was bigger and/or more complex, then maybe Ninja would bring more improvements, but just for the automatic core count usage it's not really worth using over the native tool when you already need the core count anyway to efficiently run `ctest`.

Reply:
To get this right, build and test:

```sh
JOBS=4

cmake -S test -B build
cmake --build build --config Release -j $JOBS
ctest -C Release -j ${JOBS} --output-on-failure
```

Reply:
Here's the [hardware specification](https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources) for GitHub provided runners if you want to hardcode the core numbers.

Reply:
@friendlyanon the ctest call mentioned above isn't right I guess:

```
Test project /__w/FunctionalPlus/FunctionalPlus
No tests were found!!!
```

Reply:
So better pass both – `--config Release` and `-D CMAKE_BUILD_TYPE=Release`?

Reply:
`ctest` unfortunately doesn't have a handy directory specifier like `cmake`'s `-B` and `--build`, it has to be run from the build directory, i.e. you need to `cd` first.

Reply:
I see, adding a `cd` solved the problem.